python/test/unit_distributed/test_contexts_mpi4py.py

```diff
@@ -42,14 +42,31 @@ def test_communicator_mpi4py(self):
     def test_context_mpi4py(self):
         comm = arb.mpi_comm_from_mpi4py(mpi.COMM_WORLD)
 
-        # test context with mpi usage
+        # test context with mpi
+        ctx = arb.context(mpi=comm)
```

Comment:
A more user-friendly approach would be to support mpi4py directly, to that this
```python
comm = arb.mpi_comm_from_mpi4py(mpi.COMM_WORLD)
ctx = arb.context(mpi=comm)
```

would become
```python
ctx = arb.context(mpi.COMM_WORLD)
```
How do you feel about this?

Reply:
I think this would be good, but using `ctx = arb.context(mpi=mpi.COMM_WORLD)` as the code is right now gives (the correct) `RuntimeError: mpi must be None, or an MPI communicator.`, since it is not a `mpi_comm_shim`.

Reply:
Indeed. We would need to extend the interface to support this.